pkg/server/endpoints/registration/handler.go

```diff
@@ -545,6 +547,95 @@ func (h *Handler) ListAgents(ctx context.Context, listReq *registration.ListAgen
 	return &registration.ListAgentsResponse{Nodes: resp.Nodes}, nil
 }
 
+func (h *Handler) MintX509SVID(ctx context.Context, req *registration.MintX509SVIDRequest) (_ *registration.MintX509SVIDResponse, err error) {
```

Comment:
What behavior should users expect if the requested TTL exceeds the (1) the server's max configured TTL (e.g. 24 hours), and (2) the server's max _effective_ TTL (e.g. 4 hours).

1. In the case of exceeding the server's max configured TTL, it could make sense to throw an error, since this is clearly not allowed, and no user should reasonably expect to run `mint x509 -ttl 100h` when the server TTL is 24h.

2. In the case of exceeding the server's max _effective_ TTL, it's less clear to me. A user could reasonably expect to run `mint x509 -ttl 12h` if the server TTL is 24h. However, it's likely that the effective TTL of the minted x509 could be clamped to 4h (24/6). Do we just advise users to check the TTL in the response? Do they have any recourse if the TTL is less than they require?

Reply:
1: I think it would be fine to fail in this condition
2: I don't think we should fail here. Certificates returned from the workload API are in the same boat. The TTL on the registration entry is really a "max ttl" and callers can't expect any specific TTL.

Maybe it makes sense to rename ttl field to max_ttl for these minting RPCs, which might change the expectation.

Reply:
Sounds good - either `ttl` or `max_ttl` is fine for naming, but at least documenting that in the proto will be a good step